<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Pathway analysis by aligator — aligator • snpGeneSets</title><script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Pathway analysis by aligator — aligator"><meta name="description" content="This function performs pathway analysis for every pathway at a particular MSigDB gene-set
category using the ALIGATOR algorithm, and returns pathway-specific unadjusted p-value and
adjusted p-value by permutation tests."><meta property="og:description" content="This function performs pathway analysis for every pathway at a particular MSigDB gene-set
category using the ALIGATOR algorithm, and returns pathway-specific unadjusted p-value and
adjusted p-value by permutation tests."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">snpGeneSets</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch"><li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Pathway analysis by aligator</h1>

      <div class="d-none name"><code>aligator.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function performs pathway analysis for every pathway at a particular MSigDB gene-set
category using the ALIGATOR algorithm, and returns pathway-specific unadjusted p-value and
adjusted p-value by permutation tests.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">aligator</span><span class="op">(</span><span class="va">snpGeneP</span>,setType<span class="op">=</span><span class="fl">2</span>, </span>
<span>    Nsample<span class="op">=</span><span class="fl">5000</span>, Btimes<span class="op">=</span><span class="fl">1000</span>, pcut<span class="op">=</span><span class="fl">0.05</span>, seed<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <p></p>
<dl><dt id="arg-snpgenep">snpGeneP<a class="anchor" aria-label="anchor" href="#arg-snpgenep"></a></dt>
<dd><p>It specifies a data frame containing columns of 'snp', 'gene_id' and 'p', which
	lists all analyzed SNPs, SNP-mapped genes  and GWAS SNP association p-value</p></dd>

<dt id="arg-settype">setType<a class="anchor" aria-label="anchor" href="#arg-settype"></a></dt>
<dd><p>'setType' is a integer or numeric value of gene-set category at [0,20]. It specifies the gene
	set category of MSigDB, from which every gene set is tested for enrichment.
	The setType takes default value of 2 ("C2: curated gene sets").  The complete list of gene set
	category can be obtained by 'getSetType' function.</p></dd>

  <dt id="arg-nsample">Nsample<a class="anchor" aria-label="anchor" href="#arg-nsample"></a></dt>
<dd><p>The number of random samples generated for calculating pathway-specific unadjusted p-value</p></dd>

  <dt id="arg-btimes">Btimes<a class="anchor" aria-label="anchor" href="#arg-btimes"></a></dt>
<dd><p>The number of bootstrap procedures for conducting permutation tests to get pathway adjusted p-value</p></dd>

  <dt id="arg-pcut">pcut<a class="anchor" aria-label="anchor" href="#arg-pcut"></a></dt>
<dd><p>A numerical value. It defines the cut-point p-value of SNP association for selecting 'significant'
	SNPs in pathway test.</p></dd>

  <dt id="arg-seed">seed<a class="anchor" aria-label="anchor" href="#arg-seed"></a></dt>
<dd><p>The seed number for ALIGATOR-based pathway test</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>This pathway analysis is based on ALIGATOR algorithm. It takes a data frame of 'snpGeneP' that
	contains a character vector 'snp', an integer/numeric vector 'gene_id' of SNP-mapped genes (Entrez
	gene ID) and a numeric vector 'p' of GWAS SNP association p-value. The analysis uses a preselected
	p-value threshold ('pcut') to select significant SNPs for testing enrichment of the mapped genes.
	The pathway-specific unadjusted p-value is calculated from randomly generated 'Nsample' samples, and
	pathway adjusted p-value is obtained by 'Btimes' bootstrap-based permutations.</p>
    </div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>The returned value is a data frame. It contains a integer vector 'pid' of pathway ID,
	a numeric vector 'p' of pathway-specific unadjusted p-value and a numeric vector 'adj_p'
	of pathway adjusted p-value by permutations.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Holmans, P., et al. (2009). "Gene ontology analysis of GWA study data sets provides insights into
the biology of bipolar disorder." Am J Hum Genet 85(1): 13-24.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>H. Mei; L Li</p>
    </div>
    <div class="section level2">
    <h2 id="note">Note<a class="anchor" aria-label="anchor" href="#note"></a></h2>


    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index">
</div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># data(T2DGWAS)</span></span></span>
<span class="r-in"><span><span class="co"># path0=aligator(snpGeneP,Nsample=100, Btimes=10)</span></span></span>
<span class="r-in"><span><span class="co"># head(path0)</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by H. Mei.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

