<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Gene-set enrichment test II based on genome-wide study (GWS) results — enrichTest2 • snpGeneSets</title><script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Gene-set enrichment test II based on genome-wide study (GWS) results — enrichTest2"><meta name="description" content="The enrichment analysis II applies the Uniform-Score Gene-Set Analysis (USGSA) method to test pathway
	enrichment based on genome-wide association or expression results."><meta property="og:description" content="The enrichment analysis II applies the Uniform-Score Gene-Set Analysis (USGSA) method to test pathway
	enrichment based on genome-wide association or expression results."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">snpGeneSets</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch"><li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Gene-set enrichment test II based on genome-wide study (GWS) results</h1>

      <div class="d-none name"><code>enrichTest2.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>The enrichment analysis II applies the Uniform-Score Gene-Set Analysis (USGSA) method to test pathway
	enrichment based on genome-wide association or expression results.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">enrichTest2</span><span class="op">(</span><span class="va">geneDF</span>, cut_score<span class="op">=</span><span class="fl">0.05</span>, hitSize<span class="op">=</span><span class="fl">0</span>, setType<span class="op">=</span><span class="fl">2</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <p></p>
<dl><dt id="arg-genedf">geneDF<a class="anchor" aria-label="anchor" href="#arg-genedf"></a></dt>
<dd><p>A data frame, matrix or array with at least two columns: 'gene_id' column of Entrez gene ID
	and 'score' column for gene measures of effect. The gene measure can be minP, 2ndP, simP and fishP
	for genome-wide association study (GWAS) and differential expression p-value for genome-wide
	expression study (GWES). The gene measure can be converted to uniform score by 'uscore' function.</p></dd>

  <dt id="arg-cut-score">cut_score<a class="anchor" aria-label="anchor" href="#arg-cut-score"></a></dt>
<dd><p>The cut point value of score that defines the significant gene. The default cut point is 0.05.
	If 'us'=TRUE, the 'score' column of 'geneDF' will be converted to uniform score first and then
	be used for defining significant genes. If 'us'=FALSE, the 'score' column of 'geneDF' will be
	directly used for defining significant genes.</p></dd>


  <dt id="arg-hitsize">hitSize<a class="anchor" aria-label="anchor" href="#arg-hitsize"></a></dt>
<dd><p>The 'hitSize' defines the cut point of filtering the pathway for enrichment test II.
	When a pathway or gene set  have &lt; 'hitSize' GWS genes of geneDF, this gene set/pathway
	will not be entered in the analysis procedure for calculating the p-value and, a NA value
	will instead be assigned.</p></dd>


  <dt id="arg-settype">setType<a class="anchor" aria-label="anchor" href="#arg-settype"></a></dt>
<dd><p>'setType' is a integer or numeric value of gene-set category at [0,20]. It specifies the gene
	set category of MSigDB, from which every gene set is tested for enrichment. The setType takes
	default value of 2 ("C2: curated gene sets"). The complete list of gene
	set category can be obtained by 'getSetType' function.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>For a GWS, e.g. genome-wide association study (GWAS) and genome-wide expression study (GWES), the gene
	effect can be measured by SNP association (GWAS) or differential gene expression (GWES). The enrichment
	test II applies Uniform-Score Gene-Set Analysis (USGSA) method that tests pathway enrichment for GWS
	significant genes. The significant genes are identified through their scores of measuring gene effects.</p>	
<p>The 'enrichTest2' function takes the argument of 'geneDF' that contains entrez gene ID ('gene_id') and
	gene score ('score'), and applies the hypergeometric test to calculate the empirical p-value of pathway
	enrichment for GWS significant genes that have the transformed uniform scores &lt; or = 'cut_score'.</p>	
<p>Gene scores, i.e. the 'score' column of 'geneDF', will be first converted to uniform scores and a gene is
	defined as significant if its uniform score is less than or equal to user pre-defined threshold of
	'cut_score'. The 'score' of geneDF can be any gene measure, minp, sndp, simp and fishp,that are
	calculated from SNP associaiton p-value by 'getGeneMeasure' function. For GWES, the 'score' can also
	be the differential expression p-value for GWES gene. In general, any gene measure with smaller value
	indicating a larger effect can be used to fill 'score' column of 'geneDF' and perform type II enrichment
	analysis.</p>
    </div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>The returned value is a list</p>
<dl><dt>enrich_test </dt>
<dd><p>A data frame of enrichment test II for every gene set/pathway of
	the specified gene-set category ('setType').</p><ul><li><p>'pid': The first column ('pid') is the MSigDB gene-set ID.</p></li>
<li><p>'size': the second column ('size') shows the number of genes that a MSigDB gene set has;</p></li>
<li><p>'genes': the third column ('genes') shows the number of GWS genes from 'geneDF' that
				belongs to the gene set for enrichment test II.</p></li>
<li><p>'sigGenes': the number of GWS significant genes that a gene set has.</p></li>
<li><p>'effect': It is the increased percentage of GWS significant genes in the pathway
				than a random expected percentage.</p></li>
<li><p>'sd': It is the standard error of 'effect'.</p></li>
<li><p>'pval': It is the gene-set enrichment p-value based on exact hypergeometric test.</p></li>
</ul><p></p></dd>

	<dt>useGenes </dt>
<dd><p>An integer vector of Entrez Gene ID from 'geneDF'  that belongs to the
			specified gene-set category ('setType') and is used for enrichment test II.</p></dd>

	<dt>nGenes </dt>
<dd><p>The number of 'geneDF' genes used for enrichment test II. This is equal
		to the length of useGenes</p></dd>

	<dt>nSigGenes </dt>
<dd><p>The number of GWS significant genes from 'geneDF' that belongs to the specified
		gene-set category ('setType')</p></dd>

	<dt>setTypeInfo</dt>
<dd><p>A list shows information of the specified gene-set category, 'setType'</p><ul><li><p>id: the gene-set category ID</p></li>
<li><p>symbol: the gene-set category symbol</p></li>
<li><p>name: the gene-set category name</p></li>
<li><p>description: the gene-set category description</p></li>
</ul><p></p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>http://www.broadinstitute.org/gsea/msigdb/collections.jsp</p>
<p>Mei, H., Li, L., Liu, S., Jiang, F., Griswold, M., and Mosley, T. (2015).
The uniform-score gene set analysis for identifying common pathways associated
with different diabetes traits. BMC genomics 16, 336. PMID: 25898945.</p>
<p>Mei, H., Li, L., et.al. (2016). snpGeneSets: An R Package for Genome-Wide Study Annotation. G3 6, 4087-4095. PMID: 27807048.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>H. Mei; L Li</p>
    </div>
    <div class="section level2">
    <h2 id="note">Note<a class="anchor" aria-label="anchor" href="#note"></a></h2>


    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="enrichTest1.html">enrichTest1</a></code>, <code><a href="enrichTest2_Perm.html">enrichTest2_Perm</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># setGenes=getSetGenes(2)</span></span></span>
<span class="r-in"><span><span class="co"># length(setGenes)</span></span></span>
<span class="r-in"><span><span class="co"># head(setGenes)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># geneDF=data.frame(gene_id=setGenes, score=runif(length(setGenes)))</span></span></span>
<span class="r-in"><span><span class="co"># enrich2_rst=enrichTest2(geneDF)</span></span></span>
<span class="r-in"><span><span class="co"># names(enrich2_rst)</span></span></span>
<span class="r-in"><span><span class="co"># head(enrich2_rst$enrich_test)</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by H. Mei.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

