<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Get permutation p-value of enrichment test II — getEnrich2P • snpGeneSets</title><script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Get permutation p-value of enrichment test II — getEnrich2P"><meta name="description" content="This function gets permutation p-value of enrichment test II from pre-generated
	distribution table of 10,000 permutations."><meta property="og:description" content="This function gets permutation p-value of enrichment test II from pre-generated
	distribution table of 10,000 permutations."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">snpGeneSets</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch"><li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Get permutation p-value of enrichment test II</h1>

      <div class="d-none name"><code>getEnrich2P.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function gets permutation p-value of enrichment test II from pre-generated
	distribution table of 10,000 permutations.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">getEnrich2P</span><span class="op">(</span>setP<span class="op">=</span><span class="cn">NULL</span>, setType<span class="op">=</span><span class="fl">2</span>, cut_score<span class="op">=</span><span class="fl">0.05</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <p></p>
<dl><dt id="arg-setp">setP<a class="anchor" aria-label="anchor" href="#arg-setp"></a></dt>
<dd><p>A numeric vector of observed pathway empirical p-value from enrichment test II by 'enrichTest2' function.
	If 'setP=NULL', the function will return permutation distribution table for a particular gene-set
	category. If 'setP' is not NULL, a permuted p-value will be calculated for every observed
	p-value of 'setP'</p></dd>

  <dt id="arg-settype">setType<a class="anchor" aria-label="anchor" href="#arg-settype"></a></dt>
<dd><p>'setType' is a integer or numeric value of gene-set category at [0,20]. It defines candidate
	gene sets for getting permutation distribution table. The permutation table is the minimum p-value
	of the candidate gene sets from every permutation. 'setType' in default is 2 and the function
	gets permutation distribution table of all gene sets from the category: "C2: curated gene sets".
	The complete list of gene set category can be obtained by 'getSetType' function.</p></dd>

<dt id="arg-cut-score">cut_score<a class="anchor" aria-label="anchor" href="#arg-cut-score"></a></dt>
<dd><p>The 'cut_score' is the cut point value of score that defines the GWAS significant gene.
	For this function, it specifies the permutation table that is pre-generated from the cut_score.
	(refer to 'enrichTest2_Perm' function for details). Currently, only two permutation tables are
	available that are based on 0.05 and 0.01. The default cut point is 0.05.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>The function will get the permutation distribution for the smallest p-value
	of candidate gene sets (defined by 'setType') from pre-generated data of 10,000
	permutations. If the 'setP'=NULL, the permutation distribution will be returned. If 'setP'
	is not NULL, a permuted p-value will be calculated for every observed p-value of 'setP' by
	comparing it with the permutation table.</p>
    </div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>The returned value is a list</p>
<dl><dt>perm </dt>
<dd><p>It is a data frame that shows the observed enrichment p-value at the first column
	('setP'), and the corresponding permutation p-value at the second column ('p') .</p></dd>

<dt>dist </dt>
<dd><p>The permutation distribution table of the smallest p-value for the
	candidate gene sets (see 'setType').</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Mei, H., Li, L., Liu, S., Jiang, F., Griswold, M., and Mosley, T. (2015).
The uniform-score gene set analysis for identifying common pathways associated
with different diabetes traits. BMC genomics 16, 336. PMID: 25898945.</p>
<p>Mei, H., Li, L., et.al. (2016). snpGeneSets: An R Package for Genome-Wide Study Annotation. G3 6, 4087-4095. PMID: 27807048.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>H. Mei; L Li</p>
    </div>
    <div class="section level2">
    <h2 id="note">Note<a class="anchor" aria-label="anchor" href="#note"></a></h2>


    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="enrichTest2_Perm.html">enrichTest2_Perm</a></code>, <code><a href="enrichTest2.html">enrichTest2</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># test1=getEnrich2P(setP=1e-05)</span></span></span>
<span class="r-in"><span><span class="co"># names(test1)</span></span></span>
<span class="r-in"><span><span class="co"># test1$perm</span></span></span>
<span class="r-in"><span><span class="co"># class(test1$dist)</span></span></span>
<span class="r-in"><span><span class="co"># head(test1$dist)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># test1_01=getEnrich2P(setP=1e-05, cut_score=0.01)</span></span></span>
<span class="r-in"><span><span class="co"># test1_01$perm</span></span></span>
<span class="r-in"><span><span class="co"># head(test1_01$dist)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># test2=getEnrich2P(setP=1e-05, setType=0)</span></span></span>
<span class="r-in"><span><span class="co"># test2$perm</span></span></span>
<span class="r-in"><span><span class="co"># head(test2$dist)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># test2_01=getEnrich2P(setP=1e-05, setType=0, cut_score=0.01)</span></span></span>
<span class="r-in"><span><span class="co"># test2_01$perm</span></span></span>
<span class="r-in"><span><span class="co"># head(test2_01$dist)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># test3=getEnrich2P(setP=c(1e-6, 1e-5, 1e-4,1e-3, 0.01, 0.05))</span></span></span>
<span class="r-in"><span><span class="co"># test3$perm</span></span></span>
<span class="r-in"><span><span class="co"># head(test3$dist)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># test3_01=getEnrich2P(setP=c(1e-6, 1e-5, 1e-4,1e-3, 0.01, 0.05), cut_score=0.01)</span></span></span>
<span class="r-in"><span><span class="co"># test3_01$perm</span></span></span>
<span class="r-in"><span><span class="co"># head(test3_01$dist)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># test4=getEnrich2P()</span></span></span>
<span class="r-in"><span><span class="co"># head(test4$dist)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># test4_01=getEnrich2P(cut_score=0.01)</span></span></span>
<span class="r-in"><span><span class="co"># head(test4_01$dist)</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by H. Mei.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

