<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Get SNPs in a genomic region — getRegionSNP • snpGeneSets</title><script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Get SNPs in a genomic region — getRegionSNP"><meta name="description" content="The function identifies SNPs within a candidate genomic region from NCBI Gene database based
on genome build GRCh37/hg19 or GRCh38/hg38"><meta property="og:description" content="The function identifies SNPs within a candidate genomic region from NCBI Gene database based
on genome build GRCh37/hg19 or GRCh38/hg38"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">snpGeneSets</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch"><li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Get SNPs in a genomic region</h1>

      <div class="d-none name"><code>getRegionSNP.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>The function identifies SNPs within a candidate genomic region from NCBI Gene database based
on genome build GRCh37/hg19 or GRCh38/hg38</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">getRegionSNP</span><span class="op">(</span><span class="va">regionDF</span>, GRCh<span class="op">=</span><span class="fl">37</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <p></p>
<dl><dt id="arg-regiondf">regionDF<a class="anchor" aria-label="anchor" href="#arg-regiondf"></a></dt>
<dd><p>It is a data frame. It must contain the columns of 'chr' and 'start'. The 'chr' is a
	character vector of chromosomes. The supported chromosome ID can be obtained by function,
	'getChrs'. The 'start' is a integer or numeric vector of region start position.
	An optional column is 'end'. If 'end' exists, a genomic region will be defined by 'start'
	and 'end'. Otherwise, the genomic region will be the single nucleotide position defined by
	'start'. The 'start' and 'end' position are based on unit of base pairs.</p></dd>

  <dt id="arg-grch">GRCh<a class="anchor" aria-label="anchor" href="#arg-grch"></a></dt>
<dd><p>The reference genome build number, either 37 or 38. If the number is 37, it is
GRCh37 or hg19; and if the number is 38, it is GRCh38 or hg38</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>The function looks for variants in a genomic region that have a unique position from
	NCBI dbSNP and 1000 Genome. SNPs mapping in more than one place are not included.
	If regionDF contains only 'chr' and 'start', the function will identify the SNPs at the
	defined 'start' position of 'chr'. if regionDF contains 'chr',  'start' and 'end',
	the function will identify all SNPs at the specified genomic regions.</p>
    </div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>The returned value is a data frame.</p>
<dl><dt>chr </dt>
<dd><p>The chromosome ID of the identified SNP</p></dd>

<dt>pos </dt>
<dd><p>variant position in base pair</p></dd>

<dt>snp </dt>
<dd><p>SNP rs id</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>http://www.1000genomes.org/; http://www.ncbi.nlm.nih.gov/SNP/</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>H Mei; L Li</p>
    </div>
    <div class="section level2">
    <h2 id="note">Note<a class="anchor" aria-label="anchor" href="#note"></a></h2>



    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="getSNPMap.html">getSNPMap</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">#Get SNP at 78786077 of chr 14.</span></span></span>
<span class="r-in"><span><span class="co"># regionDF=data.frame(chr="14", start=78786077, stringsAsFactors=FALSE);</span></span></span>
<span class="r-in"><span><span class="co"># getRegionSNP(regionDF)</span></span></span>
<span class="r-in"><span><span class="co"># getRegionSNP(regionDF,38)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#Get SNPs at start positions of multiple chromosomes </span></span></span>
<span class="r-in"><span><span class="co"># chr=c("14","1","18","16","16");</span></span></span>
<span class="r-in"><span><span class="co"># start=c(78786077, 213910494, 57850422, 53813450, 53820527);</span></span></span>
<span class="r-in"><span><span class="co"># end=start+1000;</span></span></span>
<span class="r-in"><span><span class="co"># regionDF=data.frame(chr=chr, start=start, stringsAsFactors=FALSE);</span></span></span>
<span class="r-in"><span><span class="co"># getRegionSNP(regionDF)</span></span></span>
<span class="r-in"><span><span class="co"># getRegionSNP(regionDF,38)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#get all SNPs at multiple genomic regions defined by 'start' and 'end'</span></span></span>
<span class="r-in"><span><span class="co"># regionDF=data.frame(chr=chr, start=start, end=end, stringsAsFactors=FALSE);</span></span></span>
<span class="r-in"><span><span class="co"># getRegionSNP(regionDF)</span></span></span>
<span class="r-in"><span><span class="co"># getRegionSNP(regionDF,38)</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by H. Mei.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

